### ISSUE TITLE
[cdk-pipelines]: ManualApproval after deploy blocks inbound execution before deploy

### ISSUE THREAD

**Issue by mh-love** [aws/aws-cdk#13806]
ManualApproval at the end of the CDK app stage - _after_ deploy - blocks inbound executions.

### Reproduction Steps

```
import * as codepipeline from '@aws-cdk/aws-codepipeline';
import * as codepipeline_actions from '@aws-cdk/aws-codepipeline-actions';
import { CdkPipeline, SimpleSynthAction, ShellScriptAction } from '@aws-cdk/pipelines';

import * as CDK from '@aws-cdk/core';
import { MyStack } from './MyStack';

class MyStackStage extends CDK.Stage {
  constructor(scope: CDK.Construct, id: string) {
    super(scope, id);

    new MyStack(this, 'MyStack');
  }
}

export class PipelineStack extends CDK.Stack {
  constructor(scope: CDK.Construct, id: string) {
    super(scope, id);

    const sourceArtifact = new codepipeline.Artifact();
    const cloudAssemblyArtifact = new codepipeline.Artifact();

    const sourceAction = new codepipeline_actions.BitBucketSourceAction({
      actionName: 'GitHubV2',
      output: sourceArtifact,
      connectionArn: 'connectionArn',
      owner: 'owner',
      repo: 'repo',
      branch: 'branch'
    });

    const cdkPipeline = new CdkPipeline(this, 'CdkPipeline', {
      pipelineName: 'Pipeline',
      crossAccountKeys: false,
      cloudAssemblyArtifact,
      sourceAction,
      synthAction: SimpleSynthAction.standardNpmSynth({
        sourceArtifact,
        cloudAssemblyArtifact,
        environment: { privileged: true },
        installCommand: 'make install',
        buildCommand: 'make build && make tests',
        synthCommand: 'cdk deploy --context stage=dev MyStack -v'
      })
    });

    const app = new MyStackStage(this, 'MyStackDev');

    const cdkPipelineStage = cdkPipeline.addApplicationStage(app);
    
    cdkPipelineStage.addActions(new ShellScriptAction({
      actionName: 'ApresDeploy',
      additionalArtifacts: [sourceArtifact],
      commands: ['make apres-deploy'],
      runOrder: 100
    }));

    cdkPipelineStage.addActions(new ShellScriptAction({
      actionName: 'IntegrationTests',
      additionalArtifacts: [sourceArtifact],
      commands: ['make integration-tests'],
      runOrder: 101
    }));

    cdkPipelineStage.addActions(new codepipeline_actions.ManualApprovalAction({
      actionName: 'ManualApproval',
      runOrder: 102
    }));

    cdkPipelineStage.addActions(new ShellScriptAction({
      actionName: 'PullRequest',
      additionalArtifacts: [sourceArtifact],
      commands: ['make pull-request'],
      runOrder: 103
    }));
  }
}
```

### What did you expect to happen?

When using the ManualApproval action, I have to manually approve deployments, even though it seems like flow-wise this action comes after the deploy. I'm trying to achieve "dev" stage deployments that deploy w/o intervention and a manual approval step is only required for the next action, which is a pull request to trigger the next stage's pipeline. 

### What actually happened?

I would expect ManualApproval to _not_ block incoming execution id "ad8..." and to just block the "PullRequest" action (it does both). Since "2a5..." wasn't approved to go to the next "PullRequest" action, ManualApproval _does_ block incoming execution id "ad8" from deploying.

### Environment

  - **CDK CLI Version  :**  1.90.0 (build 7edba31)
  - **Framework Version:** 1.90.0 
  - **Node.js Version:**  v12.18.3
  - **OS               :** Catalina 10.15.7
  - **Language (Version):**  TypeScript (3.8.3)

### Other

--- 

This is :bug: Bug Report
<img width="272" alt="Screen Shot 2021-03-25 at 3 18 45 PM" src="https://user-images.githubusercontent.com/77029949/112551159-c5916280-8d7d-11eb-87ef-6e291394a4c6.png">

**Comment by rix0rrr** [comment]
Right. Looks like the only proper solution to that is to add the manual approval into its own stage, then.

**Comment by rix0rrr** [comment]
Will be addressed by #12326, closing early as duplicate to keep tracking more manageable.
